server/src/test/java/org/opensearch/cluster/coordination/CoordinationStateTests.java

```diff
@@ -886,8 +898,69 @@ public void testSafety() {
         ).runRandomly();
     }
 
-    public static CoordinationState createCoordinationState(PersistedState storage, DiscoveryNode localNode) {
-        return new CoordinationState(localNode, storage, ElectionStrategy.DEFAULT_INSTANCE);
+    public void testHandlePrePublishAndCommitWhenRemoteStateDisabled() {
+        final RemoteClusterStateService remoteClusterStateService = Mockito.mock(RemoteClusterStateService.class);
+
+        final PersistedState remotePersistedState = new RemotePersistedState(remoteClusterStateService);
+        final CoordinationState coordinationState = createCoordinationState(psr1, node1, Settings.EMPTY);
+        final VotingConfiguration initialConfig = VotingConfiguration.of(node1);
+        final ClusterState clusterState = clusterState(0L, 0L, node1, initialConfig, initialConfig, 42L);
+        coordinationState.handlePrePublish(clusterState);
```

Comment:
in this assertion would be no NPE and there is no remotePersistedState in registry.